src/adapters/TelnetAdapter.ts

```diff
@@ -543,13 +543,24 @@ export class TelnetAdapter {
 
                 //write a for loop to print every value from the array. This gets around the `...` after the 100th item issue in the roku print call
             } else if (['roarray', 'rolist', 'roxmllist', 'robytearray'].includes(lowerExpressionType)) {
-                data = await this.requestPipeline.executeCommand(
-                    `for each vscodeLoopItem in ${expression} : print "vscode_is_string:"; (invalid <> GetInterface(vscodeLoopItem, "ifString")); vscodeLoopItem : end for`
-                    , true);
+                const command = [
+                    `for each vscodeLoopItem in ${expression} : print ` +
+                    `   "vscode_type_start:" + type(vscodeLoopItem) + ":vscode_type_stop "`,
+                    `   "vscode_is_string:"; (invalid <> GetInterface(vscodeLoopItem, "ifString"))`,
+                    `   vscodeLoopItem :` +
+                    ` end for`
+                ].join(';');
+                data = await this.requestPipeline.executeCommand(command, true);
             } else if (['roassociativearray', 'rosgnode'].includes(lowerExpressionType)) {
-                data = await this.requestPipeline.executeCommand(
-                    `for each vscodeLoopKey in ${expression}.keys() : print "vscode_key_start:" + vscodeLoopKey + ":vscode_key_stop " + "vscode_is_string:"; (invalid <> GetInterface(${expression}[vscodeLoopKey], "ifString")); ${expression}[vscodeLoopKey] : end for`,
-                    true);
+                const command = [
+                    `for each vscodeLoopKey in ${expression}.keys(): print` +
+                    `   "vscode_key_start:" + vscodeLoopKey + ":vscode_key_stop "`,
+                    `   "vscode_type_start:" + type(${expression}[vscodeLoopKey]) + ":vscode_type_stop "`,
```

Comment:
This is the new line I added here (and to the similar code above). 

We had typescript code to look up the variable types, but it was getting the types wrong most of the time. This guarantees we get the correct types since we ask the device for them.